lavalink/client.py

```diff
@@ -39,26 +36,44 @@ class Client:
         A dictionary representing region -> discord endpoint. You should only
         change this if you know what you're doing and want more control over
         which regions handle specific locations.
+    connect_back: Optional[bool]
+        A boolean that determines if a player will connect back to the
+        node it was originally connected to. This is not recommended to do since
+        the player will most likely be performing better in the new node.
     """
 
     def __init__(self, user_id: int, shard_count: int = 1,
-                 loop=None, player=DefaultPlayer, regions: dict = None):
+                 loop=None, player=DefaultPlayer, regions: dict = None, connect_back: bool = False):
         self._user_id = str(user_id)
+        self._connect_back = connect_back
         self._shard_count = str(shard_count)
         self._loop = loop or asyncio.get_event_loop()
         self.node_manager = NodeManager(self, regions)
         self.players = PlayerManager(self, player)
 
-        self._event_hooks = []
+        self._event_hooks = {}
 
         self._session = aiohttp.ClientSession(
             connector=aiohttp.TCPConnector(loop=loop),
             timeout=aiohttp.ClientTimeout(total=30)
         )  # This session will be used for websocket and http requests.
 
-    def add_event_hook(self, hook):
-        if hook not in self._event_hooks:
-            self._event_hooks.append(hook)
+    def add_event_hook(self, hook, event: Event = None):
+        """
+
+        :param hook:
+            The hook to be added, that will be dispatched when an event is dispatched.
+            If `event` parameter is left empty, then it will run when any event is dispatched.
+        :param event:
+            The event the hook belongs to. This will dispatch when that specific event is
+            dispatched.
+        """
+        event_hooks = self._event_hooks.get(event)
+        if hook not in event_hooks:
+            if event_hooks is None:
```

Comment:
@Devoxin what's the more performant version of this?

Reply:
Wdym?

Reply:
Oh, you mean if statement? `if not event_hooks`

Reply:
Still not improved here @SoulSen :^)

Reply:
Not sure of any performant way though.

Reply:
`if not event_hooks` instead of `if event_hooks is None` :^)